File: 101. Symmetric Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        return self.mirror(root,root)
    def mirror(self,t1,t2):
        if(t1==None and t2==None):
            return True
        elif(t1==None or t2==None):
            return False
        return (t1.val==t2.val) and self.mirror(t1.left,t2.right) and self.mirror(t1.right,t2.left)

File: 101_Symmetric_Tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool ismirror(TreeNode* left_node,TreeNode* right_node)
    {
        if(left_node==NULL&&right_node==NULL)
        {
            return true;
        }
        else if(left_node==NULL || right_node==NULL)
        {
            return false;
        }
        return (left_node->val==right_node->val && ismirror(left_node->left,right_node->right) && ismirror(left_node->right,right_node->left) );
    }
    bool isSymmetric(TreeNode* root) {
        return ismirror(root,root);
    }
};